File: goblin-cache/src/main/java/org/goblinframework/cache/bean/GoblinCacheMethod.java
package org.goblinframework.cache.bean;

import java.util.LinkedList;
import java.util.List;

public class GoblinCacheMethod {

  public Class<?> type;

  public final List<GoblinCacheMethodParameter> parameterList = new LinkedList<>();
  public int annotationCount = 0;
  public int multipleCount = 0;
  public int multipleIndex = -1;
  public int parameterCount = 0;

}

File: goblin-cache/src/main/java/org/goblinframework/cache/bean/GoblinCacheMethod.kt
package org.goblinframework.cache.bean

class GoblinCacheMethod internal constructor() {

  internal var type: Class<*>? = null
  internal var parameterList: List<GoblinCacheMethodParameter>? = null
  internal var annotationCount = 0
  internal var multipleCount = 0
  internal var multipleIndex = -1
  internal var parameterCount = 0

}